### PR TITLE
 [feature] Allow users to interactively configure their aws config

### PR DESCRIPTION
**Pull request by edulop91**
screen

still missing some pieces here and there, but thought this might be a nicer interface to let users generate their aws config. going to follow up with a pr to clean this up and actually enable this to work server-side.

To play around with it you can pull down this branch and then `go run main.go configure`

### CODE REVIEW

File: pkg/aws_config_server/shared_variables_test.go [hunk 1]
```diff
@@ -205,51 +206,66 @@ var alternatePolicyDocument = &AlternatePolicyDocument{
...
 	{
-		Arn:      aws.String("roleARN"),
+		Arn:      aws.String(BareRoleARN("roleARN").String()),
 		RoleName: aws.String("testRoles0"),
 	},
 }
 var testRoles1 = []*iam.Role{
 	{
-		Arn:      aws.String("roleARN0"),
+		Arn:      aws.String(BareRoleARN("roleARN0").String()),
```
Comment (crohacz) [inline pkg/aws_config_server/shared_variables_test.go:215]: I'm curious about the purpose of the BareRoleARN function here. It appears to me that the BareRoleARN function gets passed a string, then returns a pointer to an arn object. Once we receive that return value, we simply convert it back into a string. I could be misreading this though as I don't read Go very well but would be interested to learn why it's necessary here.

File: pkg/aws_config_client/completer_test.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+		} else {
+			r.Error(err)
+			r.Equal(test.err.Error(), err.Error())
+		}
+
+	}
+}
+
+// For now generate dummy data, will later on use this for tests instead
+func generateDummyData() map[server.ClientID][]server.ConfigProfile {
```
Comment (kuannie1) [inline pkg/aws_config_client/completer_test.go:74]: I don't think your tests include the case where we have one account with 2+ roles (note that there's one test id per account name here)

File: pkg/aws_config_server/list_roles.go [hunk 3]
```diff
@@ -139,10 +139,14 @@ func clientRoleMapFromProfile(ctx context.Context, acctName string, roles []*iam
...
 
+			roleARN, err := arn.Parse(*role.Arn)
+			if err != nil {
+				return errors.Wrapf(err, "could not parse arn %s", *role.Arn)
+			}
+
 			currentConfig := ConfigProfile{
-				acctName: acctName,
-				roleARN:  *role.Arn,
-				roleName: *role.RoleName,
```
Comment (kuannie1) [inline pkg/aws_config_server/list_roles.go:145]: Should we delete the roleName attribute from the ConfigProfile spec?

File: pkg/aws_config_client/completer_test.go [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+		},
+		{
+			AcctName: "foo2",
+			RoleARN: arn.ARN{
+				AccountID: "foo_id_2",
+				Resource:  "foo2RoleName",
+			},
+		},
+	}
+
```
Comment (kuannie1) [inline pkg/aws_config_client/completer_test.go:107]:
```suggestion
// Same account, different roles
configProfile3 := []server.ConfigProfile{
		{
			AcctName: "foo3",
			RoleARN: arn.ARN{
				AccountID: "foo_id_3",
				Resource:  "foo3RoleName1",
			},
		},
		{
			AcctName: "foo3",
			RoleARN: arn.ARN{
				AccountID: "foo_id_3",
				Resource:  "foo3RoleName2",
			},
		},
	}
```

File: pkg/aws_config_server/shared_variables_test.go (hunk 1 above)
Comment (edulop91) [reply pkg/aws_config_server/shared_variables_test.go:215]: I'm lazy and didn't want to figure out how to write `arn:::::::roleARN0` so instead wrote a little function that would do it for me.

File: pkg/aws_config_client/completer_test.go (hunk 2 above)
Comment (edulop91) [reply pkg/aws_config_client/completer_test.go:82]: pushed an update

File: pkg/aws_config_client/completer_test.go (hunk 4 above)
Comment (edulop91) [reply pkg/aws_config_client/completer_test.go:115]: ah, didn't see this in time. Made a similar change. Thank you.

**Pull request merged**